cmake/scripts/osx/ArchSetup.cmake

```diff
@@ -24,6 +24,13 @@ else()
   endif()
 endif()
 
+if(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "arm64" AND
+   CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin")
```

Comment:
No need for this really. we dont support building apple on anything other than apple hardware